### ISSUE TITLE
aci_rest doesn't store correct id in state

### ISSUE THREAD

**Issue by euphdk** [CiscoDevNet/terraform-provider-aci#150]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```plain
Terraform v0.13.5
+ provider registry.terraform.io/ciscodevnet/aci v0.5.0
```

### APIC version and APIC Platform

* V 4.2(5k) ACI Simulator

### Affected Resource(s)

* aci_rest

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    aci = {
      source = "CiscoDevNet/aci"
    }
  }
  required_version = ">= 0.13"
}

provider "aci" {
  username = "admin"
  password = "pass"
  url      = "https://acisim/"
  insecure = true
  version  = "0.5.0"
}

resource "aci_rest" "StpBpduFilter-enabled" {
  path    = "/api/node/mo/uni/infra/ifPol-StpBpduFilter-enabled.json"
  payload = <<EOF
  stpIfPol:
    attributes:
      dn: "uni/infra/ifPol-StpBpduFilter-enabled"
      name: "StpBpduFilter-enabled"
      ctrl: "bpdu-filter"
      rn: "ifPol-StpBpduFilter-enabled"
EOF
}

output "stp-bpdu" {
  value = aci_rest.StpBpduFilter-enabled.id
}
```

### Debug Output

<https://gist.github.com/euphdk/5f086814772218bada7970acf7cb8ee2>

### Expected Behavior

* STP resource created
* id / dn stored in state so it can be referenced elsewhere or used when destroying
* output should yield `stp-bpdu = uni/infra/ifPol-StpBpduFilter-enabled`

### Actual Behavior

* STP resource is created
* id / dn stored isn't stored in state
* output yields `stp-bpdu = {}`

**Output of state**:
```plain
$ terraform state show aci_rest.StpBpduFilter-enabled
# aci_rest.StpBpduFilter-enabled:
resource "aci_rest" "StpBpduFilter-enabled" {
    id      = jsonencode({})
    path    = "/api/node/mo/uni/infra/ifPol-StpBpduFilter-enabled.json"
    payload = <<~EOT
          stpIfPol:
            attributes:
              dn: "uni/infra/ifPol-StpBpduFilter-enabled"
              name: "StpBpduFilter-enabled"
              ctrl: "bpdu-filter"
              rn: "ifPol-StpBpduFilter-enabled"
    EOT
}
```

### Steps to Reproduce

1. `terraform apply`

**Comment by nkatarmal-crest** [comment]
@euphdk You have to give class name as an argument as it is used to extract DN. So try after giving class name into same definition and let me know if you need anything else.

**Comment by euphdk** [comment]
@nkatarmal-crest Thanks, that helped :)

Now I just encountered that it puts the value en qoutes, so it gets "double qouted", and errors out when I try to use it elsewhere?

New config:

```
resource "aci_rest" "StpBpduFilter-enabled" {
  path       = "/api/node/mo/uni/infra/ifPol-StpBpduFilter-enabled.json"
  class_name = "stpIfPol"
  payload    = <<EOF
  stpIfPol:
    attributes:
      dn: "uni/infra/ifPol-StpBpduFilter-enabled"
      name: "StpBpduFilter-enabled"
      ctrl: "bpdu-filter"
      rn: "ifPol-StpBpduFilter-enabled"
EOF
}
```

Output:

```
stp-bpdu = "uni/infra/ifPol-StpBpduFilter-enabled"
```

And state:

```
# aci_rest.StpBpduFilter-enabled:
resource "aci_rest" "StpBpduFilter-enabled" {
    class_name = "stpIfPol"
    id         = "\"uni/infra/ifPol-StpBpduFilter-enabled\""
    path       = "/api/node/mo/uni/infra/ifPol-StpBpduFilter-enabled.json"
    payload    = <<~EOT
          stpIfPol:
            attributes:
              dn: "uni/infra/ifPol-StpBpduFilter-enabled"
              name: "StpBpduFilter-enabled"
              ctrl: "bpdu-filter"
              rn: "ifPol-StpBpduFilter-enabled"
    EOT
}
```

**Comment by nkatarmal-crest** [comment]
This issue has been fixed in the provider version 0.5.1.

**Comment by euphdk** [comment]
Works like a charm 👍
